lib/configuration.ex

```diff
@@ -67,23 +86,32 @@ defmodule Configuration do
           potential_mark = get_random_letter()
           if potential_mark == player, do: get_random_letter(), else: potential_mark
       end
-      {%Player{mark: player}, %Computer{mark: computer}}
+      {%Human{mark: player}, %Computer{mark: computer}}
   end
 
-  def get_random_letter do
-    Enum.random for n <- ?a..?z, do: << n :: utf8 >> |> String.upcase
+  def get_player_marks(player, console \\ Console,  configuration \\ Configuration) do
+    mark = player |> console.get_player_marks
+    mark
+    |> Validator.is_not_blank?
+    |> configuration.set_player_marks(player, mark)
```

Comment:
Same as my comment above about the validator. This would enable you to add multiple checks here, like `check_is_not_blank`, `check_is_not_duplicated`, `check_is_single_character`, etc.

Reply:
This way, all the validation could be in the same place, rather than here and on line 101 below.

Reply:
I agree, guard style functions would be pretty great here.  They'd require a little thinking about return values, but  would be worth it.